src/cpp/core/system/Xdg.cpp

```diff
@@ -135,6 +136,49 @@ FilePath userDataDir(
    );
 }
 
+void verifyUserDirs(
+   const boost::optional<std::string>& user,
+   const boost::optional<FilePath>& homeDir)
+{
+   auto testDir = [](const FilePath& dir, const ErrorLocation& errorLoc)
+   {
+      Error error, permError;
+      if (dir.exists())
+      {
+         error = dir.testWritePermissions();
+         if (error)
+         {
+            permError = Error(
+               error.getName(),
+               error.getCode(),
+               "Missing write permissions to " + dir.getAbsolutePath() + ". Some features may not work correctly.",
+               ERROR_LOCATION);
+
+            system::User currUser;
+            error = system::User::getUserFromIdentifier(system::username(), currUser);
+            if (!error)
+               error = dir.changeOwnership(currUser, true);
```

Comment:
On Linux you need to be root to change ownership, so I think that this is very unlikely to succeed. In the pathological case (your settings/data dir is owned by someone else), the effect here will be that we log "Unable to access [dir]: Operation not permitted" which isn't super helpful. Maybe we should just figure out who the owner is and log it if we can? Or have a different error message for the attempted ownership change?

Reply:
In the screenshots in the description you can see that you get both errors in that case. Is that sufficient?

Reply:
Yes, I think that's fine. Is there a circumstance where you _would_ expect `changeOwnership` to work? Does it work on Windows, for example?

Reply:
I'm not aware of a circumstance where this would work. I included it based on @kevinushey's suggestion in #4505 because even though I can't think of a situation where it would work I don't think it's harmful and it's possible there's some filesystem permission scenario that I'm missing where it could resolve the user's issue.

Reply:
I really can't think of one. If you want to leave it in, could you add a comment explaining the reasoning above the change ownership call to save some future head-scratching?

This PR LGTM modulo said comment!

Reply:
Done.